src/message.ts

```diff
@@ -410,44 +543,82 @@ export class Message implements Sayable {
     return this.obj.content
   }
 
-  /**
-   * @todo document me
-   */
-  public type(): MsgType {
-    return this.obj.type
-  }
+  public say(text: string, replyTo?: Contact | Contact[]): Promise<any>
 
-  /**
-   * @todo document me
-   */
-  public typeSub(): MsgType {
-    if (!this.rawObj) {
-      throw new Error('no rawObj')
-    }
-    return this.rawObj.SubMsgType
-  }
+  public say(mediaMessage: MediaMessage, replyTo?: Contact | Contact[]): Promise<any>
 
   /**
-   * @todo document me
+   * Reply a Text or Media File message to the sender.
+   *
+   * @see {@link https://github.com/Chatie/wechaty/blob/master/example/ding-dong-bot.ts|Example/ding-dong-bot}
+   * @param {(string | MediaMessage)} textOrMedia
+   * @param {(Contact|Contact[])} [replyTo]
+   * @returns {Promise<any>}
+   *
+   * @example
+   * const bot = Wechaty.instance()
+   * bot
+   * .on('message', async m => {
+   *   if (/^ding$/i.test(m.content())) {
+   *     await m.say('hello world')
+   *     console.log('Bot REPLY: hello world')
+   *     await m.say(new MediaMessage(__dirname + '/wechaty.png'))
+   *     console.log('Bot REPLY: Image')
+   *   }
+   * })
    */
-  public typeApp(): AppMsgType {
-    if (!this.rawObj) {
-      throw new Error('no rawObj')
+  public say(textOrMedia: string | MediaMessage, replyTo?: Contact|Contact[]): Promise<any> {
```

Comment:
Why did you rewrite so many original methods, could you get rid of all the unrelated changes?

Because it makes the diff shows lots of mess, and make the review becomes a hard job.